src/plugin/sdl/sdl.c

```diff
@@ -117,103 +122,49 @@ static int grab_active = 0;
 static int kbd_grab_active = 0;
 static int m_cursor_visible;
 static int initialized;
-static int pre_initialized;
+static int pre_initialized = 0;
+static int txt_initialized = 0;
 static int wait_kup;
 static int copypaste;
 
-#ifndef USE_DL_PLUGINS
-#undef X_SUPPORT
-#endif
-
-#ifdef X_SUPPORT
-#ifdef USE_DL_PLUGINS
-void *X_handle;
-#define X_load_text_font pX_load_text_font
-static int (*X_load_text_font) (Display * display, int private_dpy,
-				Window window, const char *p, int *w,
-				int *h);
-#define X_handle_text_expose pX_handle_text_expose
-static int (*X_handle_text_expose) (void);
-#define X_close_text_display pX_close_text_display
-static int (*X_close_text_display) (void);
-#define X_pre_init pX_pre_init
-static int (*X_pre_init) (void);
-#define X_register_speaker pX_register_speaker
-static void (*X_register_speaker) (Display * display);
-#define X_set_resizable pX_set_resizable
-static void (*X_set_resizable) (Display * display, Window window, int on,
-				int x_res, int y_res);
-#define X_process_key pX_process_key
-static void (*X_process_key)(Display *display, XKeyEvent *e);
-#endif
+static SDL_Color text_colors[16];
+static int pointsize;
 
 #define CONFIG_SDL_SELECTION 1
 
-Display *x11_display;
-static Window x11_window;
-
-static void preinit_x11_support(void)
-{
-#ifdef USE_DL_PLUGINS
-  void *handle = load_plugin("X");
-  assert(handle);
-  X_register_speaker = DLSYM_ASSERT(handle, "X_register_speaker");
-  X_load_text_font = DLSYM_ASSERT(handle, "X_load_text_font");
-  X_pre_init = DLSYM_ASSERT(handle, "X_pre_init");
-  X_close_text_display = DLSYM_ASSERT(handle, "X_close_text_display");
-  X_handle_text_expose = DLSYM_ASSERT(handle, "X_handle_text_expose");
-  X_set_resizable = DLSYM_ASSERT(handle, "X_set_resizable");
-  X_process_key = DLSYM_ASSERT(handle, "X_process_key");
-  X_pre_init();
-  X_handle = handle;
-#endif
-}
-
-static void init_x11_support(SDL_Window * win)
-{
-  int ret;
-  SDL_SysWMinfo info;
-  SDL_VERSION(&info.version);
-  if (SDL_GetWindowWMInfo(win, &info) && info.subsystem == SDL_SYSWM_X11) {
-    x11_display = info.info.x11.display;
-    x11_window = info.info.x11.window;
-    ret = X_load_text_font(x11_display, 1, x11_window, config.X_font,
-			   &font_width, &font_height);
-    use_bitmap_font = !ret;
-  } else {
-    use_bitmap_font = 1;
-  }
-}
-#endif				/* X_SUPPORT */
-
 static void SDL_done(void)
 {
+  if (txt_initialized) {
```

Comment:
Maybe just `!use_bitmap_font`?

Reply:
as below I removed use_bitmap_font, this also has to work in the case that sdl plugin is being used just for sound support.

Reply:
So your point seems to be to treat the text
subsystem separately, so that "initialized"
means gfx subsystem, "pre_initialized" means
"gfx or sound" and "txt_initialized" means text.
Maybe its fine, but "txt_initialized && !initialized"
should likely never be the case, because, unlike
with sound, you can't have the text subsystem
w/o gfx subsystem. And in fact it is much better
to share more things between them, like texture
and surface, for the full-screen updates.
In which case it will be like before: gfx subsystem
would just have bitmap_font and ttf_font sub-modes.
But I'll leave this to later.
If you don't think the full-screen surface and
texture should be shared between gfx and text
subsystem, then lets see.

Reply:
Okay now using `use_bitmap_font` to control this.